androidApp/src/main/java/com/example/tasky/android/common/viewmodel/MainViewModel.kt

```diff
@@ -15,8 +19,29 @@ import kotlinx.coroutines.flow.stateIn
 class MainViewModel(
     konnection: Konnection,
     private val agendaRepository: IAgendaRepository,
-) : ViewModel() {
-    val isLoginStateFlow = isLoginFlow.stateIn(scope = viewModelScope, started = SharingStarted.Eagerly, initialValue = false)
+    application: Application,
+) : AndroidViewModel(application) {
```

Comment:
I strongly discourage using the `AndroidViewModel` class because the main issue is that it introduces `Context` into your ViewModel, making it untestable in unit tests. This reliance on `Context` means you can't run straightforward unit tests and will have to resort to UI tests instead, which are larger, more resource-intensive, and slower.